experimental/class-features.md

```diff
@@ -0,0 +1,61 @@
+
+# Class Features
+
+This experimental extension covers three class features proposals: 
+[Class Fields], [Static Class Features] and [Private Methods].
+
+## PropertyDefinition
+
+```js
+interface PropertyDefinition <: Node {
+    type: "PropertyDefinition";
+    key: Expression;
+    value: Expression;
+    computed: boolean;
+    static: boolean;
+}
+```
+
+## PrivatePropertyDefinition
+
+```js
+interface PrivatePropertyDefinition <: Node {
+    type: "PrivatePropertyDefinition";
+    key: PrivateName;
+    value: Expression;
+    static: boolean;
+}
+```
+
+## PrivateMethodDefinition
+
+```js
+interface PrivateMethodDefinition <: Node {
+    type: "PrivateMethodDefinition";
+    key: PrivateName;
+    value: FunctionExpression;
+    kind: "method" | "get" | "set";
+    static: boolean;
+}
+```
+
+### PrivateName
+
+```js
+interface PrivateName <: Node {
+    type: "PrivateName";
+    id: Identifier;
+}
+
+extend interface MemberExpression {
+    property: Expression | PrivateName
+}
+```
+
+A private name refers to private class elements. For a private name `#a`, its `id.name` is `a`.
+
+When the property of a member expression is a private name, its `computed` is always `false`.
```

Comment:
This seems pretty bad. Why not just have a `PrivateMemberExpression`? The reason to unify them would be that we expect people to consider them similar for some purpose. I don't expect too much of that.

Reply:
Some reasons that I can tell

- Babel has `OptionalMemberExpression` which corresponds to [OptionalChain](https://tc39.es/ecma262/#prod-OptionalChain)
- `PrivateOptionalMemberExpression` for `obj?.#x` can be confusing
- There is no `PrivateMemberExpression` like production in the spec